presto-main/src/test/java/io/prestosql/execution/TestStateMachine.java

```diff
@@ -132,11 +133,40 @@ public void testSet()
         assertNoStateChange(stateMachine, () -> stateMachine.set(State.DINNER));
     }
 
+    @Test
+    public void testTerminalState()
+            throws Exception
+    {
+        StateMachine<Optional<Throwable>> stateMachine = new StateMachine<>("test", executor, Optional.empty(), Optional::isPresent);
+        assertEquals(stateMachine.get(), Optional.empty());
+
+        Optional<Throwable> value = Optional.of(new Throwable("This is terminal state value"));
+        Optional<Throwable> value2 = Optional.of(new Throwable("This is other terminal state value"));
+
+        assertNoStateChange(stateMachine, () -> stateMachine.compareAndSet(value2, value));
+
+        assertStateChange(stateMachine,
```

Comment:
Per code-style, put all arguments on one line or each on separate line.
So this becomes:

```suggestion
        assertStateChange(
            stateMachine,
```